scheme/parsec-tpm/evidence_handler.go

```diff
@@ -104,3 +224,184 @@ func parsecTpmLookupKey(scope, tenantID, class, instance string) string {
 
 	return u.String()
 }
+
+func evidenceAsMap(e parsectpm.Evidence) (map[string]interface{}, error) {
+	data, err := e.ToJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	var out map[string]interface{}
+	err = json.Unmarshal(data, &out)
+
+	return out, err
+}
+
+func populateAttestationResult(
+	result *ear.AttestationResult,
+	evidence map[string]interface{},
+	endorsements []Endorsements,
+) error {
+	appraisal := result.Submods[SchemeName]
+
+	// once the signature on the token is verified, we can claim the HW is
+	// authentic
+	appraisal.TrustVector.Hardware = ear.GenuineHardwareClaim
+
+	ev, err := mapAsEvidence(evidence)
+	if err != nil {
+		return fmt.Errorf("failed to map as evidence: %w", err)
+	}
+	if ev.Pat == nil {
+		return errors.New("no platform token to evaluate")
+	}
+	if ev.Kat == nil {
+		return errors.New("no key attestation information")
+	}
+	attInfo, err := ev.Pat.GetAttestationInfo()
+	if err != nil {
+		return fmt.Errorf("unable to get attestation information: %w", err)
+	}
+	if attInfo == nil {
+		return errors.New("attestation information in platform token is nil")
+	}
```

Comment:
you could wrap all this stuff in a `synChecks()` function and return `handler.BadEvidence` in one go

Reply:
As discussed, as library is doing the checking on these, so removed most of them and whatever is genuine evidence failure is mapped accordingly